test/flake.nix

```
@@ -5,19 +5,28 @@
     systems.url = "github:nix-systems/default";
     process-compose-flake.url = "github:Platonic-Systems/process-compose-flake";
     services-flake.url = "github:juspay/services-flake";
+    treefmt-nix.url = "github:numtide/treefmt-nix";
   };
   outputs = inputs:
     inputs.flake-parts.lib.mkFlake { inherit inputs; } {
       systems = import inputs.systems;
       imports = [
         inputs.process-compose-flake.flakeModule
+        inputs.treefmt-nix.flakeModule
       ];
-      perSystem = { self', pkgs, system, lib, ... }: {
+      perSystem = { self', pkgs, system, lib, config, ... }: {
         _module.args.pkgs = import inputs.nixpkgs {
           inherit system;
           # Required for elastic search
           config.allowUnfree = true;
         };
+        treefmt = {
+          projectRootFile = "flake.nix";
```

```suggestion
          projectRoot = inputs.services-flake;
          projectRootFile = "flake.nix";
```

Like this you might not need to create another check

---

sure i'll give it a try

---

@shivaraj-bh this suggestion works. Although I had another concern. I am having to define `treefmt.programs` twice. once in `dev/flake.nix` and then again in `test/flake.nix`. Is there a way I could get the programs as well from the `${inputs.services-flake}/dev/flake.nix`

---

Why not just add `services-flake` in inputs of `./dev`? and it should work without having to make any changes in `./test/flake.nix`

---

sure i'll try that only

---

If this works out you can make the change in #103

---

adding to `services-flake` to `./dev/flake.nix` works. i've pushed that change

---

> If this works out you can make the change in https://github.com/juspay/services-flake/pull/103

i missed your comment before pushing to this branch. i'll close this one and push in #103 after the build is successful here